src/main/webapp/3Dscript.js

```diff
@@ -208,3 +382,71 @@ function animate() {
   renderer.render(scene, camera);
 };
 
+
+/**
+* What to do on mouse click.
+* @param {event} event event.
+ */
+function onClick(event) {
+  /* If a cylinder was the previous thing clicked, unscale it. */
+  if (oldIndex!= -1) {
+    direction.getMatrixAt(oldIndex, instanceMatrix );
+    matrix.multiplyMatrices( instanceMatrix, scaleInverseMatrix );
+    direction.setMatrixAt( oldIndex, matrix );
+    direction.instanceMatrix.needsUpdate = true;
+    oldIndex= -1;
+  }
+
+  event.preventDefault();
+  mouse.x = ( event.clientX / window.innerWidth ) * 2 - 1;
+  mouse.y = - ( event.clientY / window.innerHeight ) * 2 + 1;
+
+  /* Get array of objects intersected by raycaster */
+  raycaster.setFromCamera( mouse, camera );
+  const intersects = raycaster.intersectObjects(scene.children, true);
+
+  /* Shows properties of first instance of intersected cylinder. */
+  for (let i=0; i < intersects.length; i++) {
+    if ( intersects[i].object.type == 'Mesh' ) {
```

Comment:
You can invert the conditional to reduce nesting:
```
if (intersection.object.type != 'Mesh') {
  continue;
}
/* ... Nominal-case code ... */
break;
```

https://docs.google.com/document/d/16SSi-rIUGxJ5-ksifO6y0InCOwgbRa3lHBLnBJ8Klrk/edit?usp=sharing